src/protocols/bdx/AsyncTransferFacilitator.cpp

```diff
@@ -63,6 +64,17 @@ void AsyncTransferFacilitator::ProcessOutputEvents()
     mTransfer.GetNextAction(outEvent);
     while (outEvent.EventType != TransferSession::OutputEventType::kNone)
     {
+
+        // If the transfer session state machine generates an event of type TransferSession::OutputEventType::kInternalError,
+        // indicating that the session is in a bad state, we do not want to process any more events and terminate the session.
+        // In that case, set mDestroySelfAfterProcessingEvents to true and break from the loop so we can call DestroySelf() to clean
+        // up.
```

Comment:
```suggestion
        // If the transfer session state machine generates an event of type TransferSession::OutputEventType::kInternalError,
        // indicating that the session is in a bad state, it will keep doing that thereafter.
        //
        // So stop trying to process events, and go ahead and destroy ourselves to clean up the transfer.
```